src/metadata_validator.py

```diff
@@ -231,14 +231,87 @@ def validate_props(self, dataRecord, model):
                 errors.extend(errs)
 
         return {VALIDATION_RESULT: STATUS_ERROR if len(errors) > 0 else STATUS_PASSED, ERRORS: errors, WARNINGS: []}
-    
-    def validate_relationship(self, dataRecord, model):
+
+    def get_parent_node_cache(self, data_record_parent_nodes):
+        parent_nodes = []
+        for parent_node in data_record_parent_nodes:
+            parent_type = parent_node.get("parentType")
+            parent_id_property = parent_node.get("parentIDPropName")
+            parent_id_value = parent_node.get("parentIDValue")
+            if parent_type and parent_id_value and parent_id_value is not None:
+                parent_nodes.append({"type": parent_type, "key": parent_id_property, "value": parent_id_value})
+        exist_parent_nodes = self.mongo_dao.search_nodes_by_type_and_value(parent_nodes)
+
+        parent_node_cache = set()
+        for node in exist_parent_nodes:
+            if node.get("props"):
+                for key, value in node["props"].items():
+                    parent_node_cache.add(tuple([node.get("nodeType"), key, value]))
+        return parent_node_cache
+
+
+    def validate_relationship(self, data_record, node_definition):
         # set default return values
-        errors = []
-        warnings = []
-        result = STATUS_PASSED
-        return {VALIDATION_RESULT: result, ERRORS: errors, WARNINGS: warnings}
-    
+        result = {"result": STATUS_ERROR, ERRORS: [], WARNINGS: []}
+        if not data_record.get("parents"):
+            result["result"] = STATUS_WARNING
+            result[WARNINGS].append(create_error(FAILED_VALIDATE_RECORDS, "Parent property does not exist or empty"))
+            return result
+
+        data_record_parent_nodes = data_record.get("parents")
+        nodes = node_definition[MODEL].get("nodes", {})
```

Comment:
the node definition passed in is an instance of Model class.  To access mode dict, use node_definition.model

Reply:
removed and integrated with the model class